mingw.invoke.h

```diff
@@ -89,20 +98,19 @@ namespace detail
     typedef Invoker<std::is_member_function_pointer<typename std::remove_reference<F>::type>::value,
                     std::is_member_object_pointer<typename std::remove_reference<F>::type>::value &&
                     (sizeof...(Args) == 1)> invoker;
-    inline static auto invoke (F&& f, Args&&... args) -> decltype(invoker::invoke(std::forward<F>(f), std::forward<Args>(args)...))
+    inline static auto invoke (F&& f, Args&&... args) ->\
+     decltype(invoker::invoke(std::forward<F>(f), std::forward<Args>(args)...))
     {
       return invoker::invoke(std::forward<F>(f), std::forward<Args>(args)...);
     }
   };
 
   template<class F, class...Args>
-  auto invoke (F&& f, Args&&... args) -> decltype(InvokeResult<F, Args...>::invoke(std::forward<F>(f), std::forward<Args>(args)...))
+  auto invoke (F&& f, Args&&... args) ->\
+   decltype(InvokeResult<F, Args...>::invoke(std::forward<F>(f), std::forward<Args>(args)...))
   {
     return InvokeResult<F, Args...>::invoke(std::forward<F>(f), std::forward<Args>(args)...);
   }
-#else
-    using std::invoke;
```

Comment:
Removing this subverts the entire intent of your pull request.

Before you removed this, the code would use the C++17 standard's `std::invoke` when compiling with C++17, and use a fallback implementation when compiling with C++11 or C++14. With your changes, `std::invoke` is only used as a fallback in an implementation of `std::invoke`; this can only add inefficiency.